service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/ApplicantController.java

```diff
@@ -24,33 +32,71 @@
 
     @Autowired
     private MapperService mapperService;
+    private final ApplicantMigrationService applicantMigrationService;
+
+    @Autowired
+    public ApplicantController(MapperService mapperService,
+                               ApplicantMigrationService applicantMigrationService) {
+        this.mapperService = mapperService;
+        this.applicantMigrationService = applicantMigrationService;
+    }
+
+    @PostMapping("/about-to-start")
+    public AboutToStartOrSubmitCallbackResponse handleAboutToStart(@RequestBody CallbackRequest callbackrequest) {
+        CaseDetails caseDetails = callbackrequest.getCaseDetails();
+
+        return applicantMigrationService.setMigratedValue(caseDetails);
+    }
 
     @SuppressWarnings("unchecked")
     @PostMapping("/mid-event")
     public AboutToStartOrSubmitCallbackResponse handleMidEvent(@RequestBody CallbackRequest callbackrequest) {
         CaseDetails caseDetails = callbackrequest.getCaseDetails();
+        ImmutableList.Builder<String> validationErrors = ImmutableList.builder();
 
-        Map<String, Object> applicantData = (Map<String, Object>) caseDetails.getData().get("applicant");
-        Applicant applicant = mapperService.mapObject(applicantData, Applicant.class);
-        List<String> validationErrors = new ArrayList<String>();
+        Map<String, Object> applicantData = (Map<String, Object>)
+            defaultIfNull(caseDetails.getData().get("applicant"), null);
+
+        if (caseDetails.getData().containsKey("applicants")) {
+            List<Map<String, Object>> migratedApplicantsObject =
+                (List<Map<String, Object>>) caseDetails.getData().get("applicants");
+
+            migratedApplicantsObject.stream()
+                .map(applicant ->
+                    mapperService.mapObject((Map<String, Object>) applicant.get("value"), Applicant.class))
+                .map(Applicant::getParty)
+                .map(PartyApplicant::getPbaNumber)
+                .filter(Objects::nonNull)
+                .forEach(pbaNumber -> {
+                    System.out.println("pbaNumber = " + pbaNumber);
+                    String newPbaNumberData = PBANumberHelper.updatePBANumber(pbaNumber);
+                    validationErrors.addAll(validatePBANumber(newPbaNumberData));
+                });
+
+            caseDetails.getData().put("applicants", migratedApplicantsObject);
 
-        if (applicant.getPbaNumber() == null || applicant.getPbaNumber().isBlank()) {
-            return AboutToStartOrSubmitCallbackResponse.builder()
-                .data(caseDetails.getData())
-                .errors(validationErrors)
-                .build();
         } else {
+            OldApplicant applicant = mapperService.mapObject(applicantData, OldApplicant.class);
+
+            if (applicant.getPbaNumber() == null || applicant.getPbaNumber().isBlank()) {
```

Comment:
.isBlank() does a null check no? the or isn't necessary

Reply:
needs to be there

Reply:
what does each bit check for?

Reply:
i dont know what way youve commented but i have to comment like this, we tried just having one and came to the conclusion you needed both @swalker125

Reply:
👍

Reply:
@swalker125 @rebeccabrennan can you let me know why you needed both?

Reply:
It just doesnt catch it without them both there.